examples/gno.land/r/gov/dao/v3/memberstore/memberstore.gno

```diff
@@ -0,0 +1,105 @@
+package memberstore
+
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/r/gov/dao"
+)
+
+var members MembersByTier
+var Tiers TiersByName
+
+const (
+	T1 = "T1"
+	T2 = "T2"
+	T3 = "T3"
+)
+
+func init() {
+	members = NewMembersByTier()
+
+	Tiers = TiersByName{avl.NewTree()}
+	Tiers.Set(T1, Tier{
+		InvitationPoints: 3,
+		Payment:          90000,
+		MinSize: func(membersByTier MembersByTier, tiersByName TiersByName) int {
+			return 70
+		},
+		MaxSize: func(membersByTier MembersByTier, tiersByName TiersByName) int {
+			return 0
+		},
+		BasePower: 3,
+		PowerHandler: func(membersByTier MembersByTier, tiersByName TiersByName) float64 {
+			return 3
+		},
+	})
+
+	Tiers.Set(T2, Tier{
+		InvitationPoints: 2,
+		Payment:          90000,
+		MaxSize: func(membersByTier MembersByTier, tiersByName TiersByName) int {
+			return membersByTier.GetTierSize(T1) * 2
+		},
+		MinSize: func(membersByTier MembersByTier, tiersByName TiersByName) int {
+			return membersByTier.GetTierSize(T1) / 4
+		},
+		BasePower: 2,
+		PowerHandler: func(membersByTier MembersByTier, tiersByName TiersByName) float64 {
+			t1ms := float64(membersByTier.GetTierSize(T1))
+			t1, _ := tiersByName.GetTier(T1)
+			t2ms := float64(membersByTier.GetTierSize(T2))
+			t2, _ := tiersByName.GetTier(T2)
+
+			t1p := t1.BasePower * t1ms
+			t2p := t2.BasePower * t2ms
+
+			// capped to 2/3 of tier 1
+			t1ptreshold := t1p * (2.0 / 3.0)
+			if t2p > t1ptreshold {
+				return t1ptreshold / t2ms
+			}
+
+			return t2.BasePower
+		},
+	})
+
+	Tiers.Set(T3, Tier{
+		InvitationPoints: 1,
+		Payment:          0,
+		MaxSize: func(membersByTier MembersByTier, tiersByName TiersByName) int {
+			return 0
+		},
+		MinSize: func(membersByTier MembersByTier, tiersByName TiersByName) int {
+			return 0
+		},
+		BasePower: 1,
+		PowerHandler: func(membersByTier MembersByTier, tiersByName TiersByName) float64 {
+			t1ms := float64(membersByTier.GetTierSize(T1))
+			t1, _ := tiersByName.GetTier(T1)
+			t3ms := float64(membersByTier.GetTierSize(T3))
+			t3, _ := tiersByName.GetTier(T3)
+
+			t1p := t1.BasePower * t1ms
+			t3p := t3.BasePower * t3ms
+
+			//capped to 1/3 of tier 1
+			t1ptreshold := t1p * (1.0 / 3.0)
+			if t3p > t1ptreshold {
+				return t1ptreshold / t3ms
+			}
+
+			return t3.BasePower
+		},
+	})
+}
+
+// Get gets the Members store
+func Get() MembersByTier {
+	realm := std.PreviousRealm().PkgPath()
+	if dao.InAllowedDAOsIndex(realm) < 0 {
+		panic("this Realm is not allowed to get the Members data: " + realm)
+	}
```

Comment:
why limiting this?

Reply:
Ok, it's because the other functions won't verify the caller; they aren't "safe objects." I think that's acceptable, but it also limits the opportunity to make the member list usable by other contracts. For example, it could allow any tiered member to access other realms easily.

An alternative approach could be to create an unrestricted getter for the list. Then, the Member could be a safe object that verifies the caller for actions such as RemoveInvitationPoint, among others.